lib/openstack_query/query_blocks/query_chainer.py

```diff
@@ -0,0 +1,92 @@
+from typing import Tuple, List, Optional, Union
+
+from enums.query.props.prop_enum import PropEnum
+from enums.query.query_presets import QueryPresetsGeneric
+from enums.query.query_types import QueryTypes
+from exceptions.query_chaining_error import QueryChainingError
+
+
+class QueryChainer:
+    """
+    Helper class to handle chaining queries together
+    """
+
+    def __init__(self, chain_mappings):
+        self._chain_mappings = chain_mappings
+
+    def get_chaining_props(self) -> List[QueryTypes]:
+        """
+        Gets a list of all supported props that can be used to chain to other queries
+        """
+        return list(self._chain_mappings.keys())
+
+    def check_prop_supported(self, prop: PropEnum) -> bool:
+        """
+        Checks whether a prop can be used as a link prop for the query
+        :param prop: prop to check if it is a chain prop for given query
+        """
+        return prop in self._chain_mappings.keys()
+
+    def get_link_props(
+        self, next_query: QueryTypes
+    ) -> Optional[Tuple[PropEnum, PropEnum]]:
+        """
+        Get the link property pair to use to convert from current query to new query
+        (if mapping exists)
+        :param next_query: next Query QueryTypes
+        """
+
+        new_query_props = list(next_query.value.get_prop_mapping())
```

Comment:
Don't convert this to a list if we're going to iterate it, a generator would work fine here too